src/ZDLMainWindow.cpp

```diff
@@ -453,19 +453,28 @@ QString ZDLMainWindow::getArgumentsString(bool native_sep)
 		}
 	}
 
-	if (zconf->hasValue("zdl.save", "skill")){
+	if (zconf->hasValue("zdl.save", "monsters")){
 		bool ok;
-		QString s_skill=zconf->getValue("zdl.save", "skill");
-		int i_skill=s_skill.toInt(&ok, 10);
-
-		if (i_skill>0&&i_skill<6) {
-			args.append(" -skill ");
-			args.append(s_skill);
-		} else if (i_skill==6) {
-			args.append(" -nomonsters");
+		int i_monsters=zconf->getValue("zdl.save", "monsters").toInt(&ok, 10);
```

Comment:
Could you also add this logic to the non _WIN32 section under the second QStringList ZDLMainWindow::getArgumentsList(). Should be the same except the arguments are added with the logical shift operator and without spaces (ex.: args << "-nomonsters";)

Reply:
Also in the non _WIN32 case toInt() can be called without any arguments, it defaults to base 10 and a nullptr for ok since we don't use it there.

Reply:
> Could you also add this logic to the non _WIN32 section under the second QStringList ZDLMainWindow::getArgumentsList(). Should be the same except the arguments are added with the logical shift operator and without spaces (ex.: args << "-nomonsters";)

Oh gosh, sorry, I totally missed that! I added the needed code for it now


> Also in the non _WIN32 case toInt() can be called without any arguments, it defaults to base 10 and a nullptr for ok since we don't use it there.

Ok, I did as you suggested simply using toInt() without parameters and went with the << operator for concatenating strings rather than going with args.append() for the "non _WIN32" section
BE WARNED: I didn't compiled the code by my side so I can't assure you everything works correctly... (I guess it does anyway)